lib/remoteproc/remoteproc.c

```diff
@@ -48,7 +48,7 @@ remoteproc_get_mem(struct remoteproc *rproc, const char *name,
 	metal_list_for_each(&rproc->mems, node) {
 		mem = metal_container_of(node, struct remoteproc_mem, node);
 		if (name) {
-			if (!strncmp(name, mem->name, strlen(name)))
+			if (!strncmp(name, mem->name, RPROC_MAX_NAME_LEN))
```

Comment:
I believe the current behavior is correct. If strlen(name) > RPROC_MAX_NAME_LEN, we will get an out-of-bounds access to mem->name.

Quoting an old BSD man page I had handy:

> The strcmp() and strncmp() functions lexicographically compare the null-ter-
     minated strings s1 and s2.

     The strncmp() function compares not more than n characters.  Because
     strncmp() is designed for comparing strings rather than binary data, charac-
     ters that appear after a `\0' character are not compared.

Reply:
@edmooring: 
The issue here is strlen(name) < RPROC_MAX_NAME_LEN
Please have a look to my comment in associated issue: 
https://github.com/OpenAMP/open-amp/issues/370#issuecomment-1099401883

Reply:
After further discussion with @arnopo, I agree that this change is an improvement.